File: 2021/day_1/part_1.rb
require "pry"

numbers = File.read("data.txt").split("\n").map(&:to_i)

result = numbers.each_cons(2).inject(0) do |increases_count, numbers_pair|
  increases_count +=1 if numbers_pair[1] > numbers_pair[0]
  increases_count
end

puts result

File: 2021/day_1/part_1.exs
defmodule Counter do
  def compare_and_count([first, second | tail], total) do
    total = if first < second, do: total + 1, else: total
    compare_and_count([second | tail], total)
  end

  def compare_and_count(_number, total) do
    total
  end
end

{:ok, content} = File.read("data.txt")
numbers        = content |> String.split("\n") |> Enum.map(&String.to_integer/1)

IO.puts Counter.compare_and_count(numbers, 0)
